Sources/SwiftParser/Availability.swift

```diff
@@ -261,44 +261,46 @@ extension Parser {
   ///     platform-version → decimal-digits
   ///     platform-version → decimal-digits '.' decimal-digits
   ///     platform-version → decimal-digits '.' decimal-digits '.' decimal-digits
-  mutating func parseVersionTuple() -> RawVersionTupleSyntax {
+  mutating func parseVersionTuple(maxComponentCount: Int = 3) -> RawVersionTupleSyntax {
     if self.at(.floatingLiteral),
       let periodIndex = self.currentToken.tokenText.firstIndex(of: UInt8(ascii: ".")),
       self.currentToken.tokenText[0..<periodIndex].allSatisfy({ Unicode.Scalar($0).isDigit })
     {
       // The lexer generates a float literal '1.2' for the major and minor version.
       // Split it into two integers if possible
       let major = self.consumePrefix(SyntaxText(rebasing: self.currentToken.tokenText[0..<periodIndex]), as: .integerLiteral)
-      let (unexpectedBeforeMinorPeriod, minorPeriod) = self.expect(.period)
-      let minor = self.expectDecimalIntegerWithoutRecovery()
-      let patchPeriod: RawTokenSyntax?
-      let patch: RawTokenSyntax?
-      if let period = self.consume(if: .period) {
-        patchPeriod = period
-        patch = self.expectDecimalIntegerWithoutRecovery()
-      } else {
-        patchPeriod = nil
-        patch = nil
+
+      var components: [RawVersionComponentSyntax] = []
+      for _ in 1..<maxComponentCount {
+        guard let period = self.consume(if: .period) else {
+          break
+        }
+        let version = self.expectDecimalIntegerWithoutRecovery()
+
+        components.append(RawVersionComponentSyntax(period: period, number: version, arena: self.arena))
       }
```

Comment:
What do you think about unifying parsing of `components` and `trailingComponents` Something like the following

```suggestion
      var components: [RawVersionComponentSyntax] = []
      for i in 1… {
        guard let period = self.consume(if: .period) else {
          break
        }
        let version = self.expectDecimalIntegerWithoutRecovery()

        let versionComponent = RawVersionComponentSyntax(period: period, number: version, arena: self.arena)
        if i < maxComponentCount {
          components.append(versionComponent)
        } else {
          trailingComponents.append(versionComponent)
        }
      }
```

Reply:
I think that looks cleaner. I will make a change on it.